# Common configuration for the mavel velocity controller

# Control
control_rate: 50.0
control_frame: "map"

throttle:
  min: 0.0
  mid: 0.5
  max: 0.9

# Maximum roll/pitch in radians
tilt_max: 0.39

# Emergency Landing Velocity
failsafe_land_vel: -0.2 # Must be negative!

# Output emergency low thrust if odometry input is lost
# (enable if the low level device has an unreliable failsafe)
failsafe_output_fatal: false

# Allows fallback to the last position received if a timeout
# occurs on the position/trajectory reference
# (can be abused to allow once-off position goals to be sent)
allow_timeout_position: false

# Data Stream Rates
min_rate:
  state:
    mav_state: 0.2
    odometry: 20.0
  reference:
    trajectory: 20.0
    triplet: 20.0
    position: 5.0
    velocity: 20.0
    acceleration: 40.0

# PID Controllers
control:
  pos:
    x:
      p: 1.25
      i: 0.0
      d: 0.5
      tau: 0.05
      min: -2.0
      max: 2.0
    y:
      p: 1.25
      i: 0.0
      d: 0.5
      tau: 0.05
      min: -2.0
      max: 2.0
    z:
      p: 1.0
      i: 0.0
      d: 0.1
      tau: 0.05
      min: -1.0
      max: 2.0
  vel:
    x:
      p: 0.09
      i: 0.02
      d: 0.04
      tau: 0.05
      min: -0.2
      max: 0.2
    y:
      p: 0.09
      i: 0.02
      d: 0.04
      tau: 0.05
      min: -0.2
      max: 0.2
    z:
      p: 0.2
      i: 0.02
      d: 0.0
      tau: 0.05
      min: -0.5
      max: 0.5

